### ISSUE TITLE
Wrong vertical position of PopupMenuButton's PopupMenu on mobile using Flutter 2.0

### ISSUE THREAD

**Issue by f69** [flutter/flutter#79791]
## Steps to Reproduce

1. Create new app: run `flutter create popup_test`.
2. Replace main.dart code with code from [Material Design doc page](https://material.io/components/menus/flutter#using-menus) available at [DartPad demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/menus/dropdown&gh_ref=develop) (see **Source code** section below)
3. Run on mobile

Notch is present in actual result screenshot only to emphasis the problem (with regular status bar height popup shift is smaller).
Popup is shifted down on the value of `MediaQuery.of(context).viewPadding.top`.
I think this is due to replacement of `MediaQuery.removePadding` by `SafeArea` widget in `buildPage` method of `_PopupMenuRoute` class in `popup_menu.dart`.

Of course, one can workaround this with `PopupMenuButton(offset: Offset(0, -MediaQuery.of(context).viewPadding.top),`, but I dont think that this is the right way.

**Expected results:** 

![image](https://user-images.githubusercontent.com/18616844/113592331-af1cbe00-963d-11eb-80f1-454d44ab1685.png)

**Actual results:** 

![image](https://user-images.githubusercontent.com/18616844/113592443-d3789a80-963d-11eb-9c78-e1833fd7fb68.png)

<details>
  <summary>Source code</summary>

```javascript
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Dropdown Menu Demo',
      debugShowCheckedModeBanner: false,
      theme: ThemeData(
        primaryColor: Colors.purple,
      ),
      home: DropdownMenuDemo(),
    );
  }
}

class DropdownMenuDemo extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Dropdown Menu Button'),
        actions: [
          PopupMenuButton(
            icon: Icon(Icons.more_vert),
            itemBuilder: (BuildContext context) => <PopupMenuEntry>[
              const PopupMenuItem(
                child: ListTile(
                  leading: Icon(Icons.add),
                  title: Text('Item 1'),
                ),
              ),
              const PopupMenuItem(
                child: ListTile(
                  leading: Icon(Icons.anchor),
                  title: Text('Item 2'),
                ),
              ),
              const PopupMenuItem(
                child: ListTile(
                  leading: Icon(Icons.article),
                  title: Text('Item 3'),
                ),
              ),
              const PopupMenuDivider(),
              const PopupMenuItem(child: Text('Item A')),
              const PopupMenuItem(child: Text('Item B')),
            ],
          ),
        ],
      ),
      body: Center(),
    );
  }
}
```
</details>

<details>
  <summary>Logs</summary>

```
[√] Flutter (Channel stable, 2.0.4, on Microsoft Windows [Version 10.0.18362.1256], locale ru-RU)
    • Flutter version 2.0.4 at f:\Programs\Develop\flutter2
    • Framework revision b1395592de (4 days ago), 2021-04-01 14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at c:\Users\ds\AppData\Local\Android\Sdk
    • Platform android-30, build-tools 29.0.2
    • ANDROID_HOME = c:\Users\ds\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] Connected device (3 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 11 (API 30) (emulator)
    • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows [Version 10.0.18362.1256]
    • Chrome (web)                 • chrome        • web-javascript • Google Chrome 89.0.4389.114

• No issues found!
```
</details>

**Comment by HansMuller** [comment]
CC @shihaohong

**Comment by xu-baolin** [comment]
Fixed by #78395

**Comment by f69** [comment]
Ok. Should I close this issue?

And some remarks regarding specs and consistency (related  #70160)
[Material design spec ](https://material.io/components/menus#dropdown-menu) says, that a dropdown menu is typically positioned below the element that generates it (and it is shown on images). I also prefer position below. But even in Android S preview we can see dropdown menu positioned **over** button in app bar (for example, in Settings: in search field and in Storage page, see below)

![image](https://user-images.githubusercontent.com/18616844/113686571-0ec3a980-96d0-11eb-93ac-8713a9adcc63.png)

![image](https://user-images.githubusercontent.com/18616844/113686619-1e42f280-96d0-11eb-8901-f20db8d0cc99.png)

**Comment by xu-baolin** [comment]
@f69, hi, You can close this if your problem has been fixed.

I really appreciate your feedback. Patch #70160 on the specification is still in processing.

**Comment by xu-baolin** [comment]
Fixed via #78395
